### DISCUSSION TITLE
Access to underlying data

### DISCUSSION THREAD

**Discussion by Ianpwest** [tvtoglu/predictr#7] [General]
Thank you for the awesome library! I'm looking to use the mle analysis to process the data, however, instead of plotting it through PlotAll I would like to use other visualization tools like Excel or PowerBI to chart the data.

Is it possible to access this raw dataset after Analysis? My code looks something like this

```
from predictr import Analysis, PlotAll

prototype_a = Analysis(df=failures_a, bounds='lrb', bounds_type='2s')
prototype_a.mle()

prototype_b = Analysis(df=failures_b, bounds='pbb', bounds_type='2s')
prototype_b.mle()

objects = {'Failures': prototype_a, 'Suspensions': prototype_b}

PlotAll(objects).mult_weibull()
```

 I appreciate any and all help. Thanks!

**Comment by tvtoglu** [comment]
Hi Ian. 
Happy to hear that you like predictr :)

Yes, you can access the whole data set which is used for plotting. The data is being stored in the attributes of your object. Let me show you an example:

We got this data and our object is called "prototype_a"
`failures = [0.4508831,  0.68564703, 0.76826143, 0.88231395, 1.48287253, 1.62876357]
prototype_a = Analysis(df=failures, bounds='fb',show=True)
prototype_a.mle()`

After initiliazing your object, you can access its attributes. To show every possible attribute just run this:
`prototype_a.__dict__`

This should return a big dictionary including all attributes that you can access:
{'df': [0.4508831, 0.68564703, 0.76826143, 0.88231395, 1.48287253, 1.62876357],
 'ds': None,
 'show': True,
 'plot_style': 'ggplot',
 'x_label': 'Time to Failure',
 'y_label': 'Unreliability',
 'plot_title': 'Weibull Probability Plot',
 ...

To get the needed data for plotting, you need the Weibull parameters:
```
prototype_a.beta
prototype_a.eta
```
If you use bias-corrections, you need to use the bias-corrected attributes. E.g. for 
```
prototype_a.beta_hrbu
prototype_a.eta_hrbu
```

The lower and upper confidence bounds:
`prototype_a.bounds_lower
prototype_a.bounds_upper`

The corresponding percentile of each bounds value is shown in 
`prototype_a.unrel`

There is no attribute for the Weibull line though. Since your Weibull function is defined by the estimated beta and eta values, you can just calculate them by using the generic Weibull function in your own code. For instance:
`
Get the x-axis values by using the weibull function and plotting the data afterwards where t is an array of values for which you wnat to compute the y-axis value:
```
def weibull_func(t, beta, eta):
        def unrel_func(x_est, beta_, eta):
            return 1 - np.exp(-(t / eta) ** beta))

y_values = weibull_func(t, prototype_a.beta, prototype_a.eta)
```
Best regards

**Reply by Ianpwest** [reply]
Thank you so much for the thorough response! This is very helpful - I can't express enough how much I appreciate the timely, and useful, response. Hopefully this helps others in the future as well. Cheers
